ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/manager/bounded/TimedTask.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.eth.manager.bounded;
+
+import java.time.Duration;
+import java.time.Instant;
+
+public interface TimedTask extends Runnable {
```

Comment:
(Optional) Consider merging this with the concrete implementation `BoundedTimedTask`.  If we don't anticipate multiple implementations, it's simpler to just have a single concrete class.

Reply:
+1 mostly in Pantheon we avoid interfaces with a single implementation (unless we need to deliberately narrow the exposed API).  That said, this interface will disappear completely when we remove the timing aspects from this PR.